app/src/main/java/ru/anisart/vv/TrackingService.kt

```diff
@@ -161,29 +161,29 @@ class TrackingService : Service() {
         notificationManager.notify(SERVICE_ID, notification.build())
     }
 
-    private fun startLocating() {
-        subscription?.dispose()
-        subscription = rxLocation.location().updates(locationRequest)
-                .map(::LatLng)
-                .distinctUntilChanged()
-                .doOnNext{
-                    track.add(it)
-                    val newTile = acquiredTiles.add(latLng2tile(it))
-                    sendBroadcast(Intent(ACTION_TRACK)
-                            .putExtra(EXTRA_NEW_TILE, newTile))
-                    preferences.edit {
-                        putString(PREFERENCE_TRACK, track.toJson())
-                        if (newTile) putString(PREFERENCE_ACQUIRED_TILES, acquiredTiles.toJson())
+        private fun startLocating() {
+            subscription?.dispose()
+            subscription = rxLocation.location().updates(locationRequest)
```

Comment:
**Reporter**: ANDROIDLINT
**Rule**: MissingPermission
**Severity**: ERROR
**File**: C:/Users/aanisimov/.jenkins/workspace/vv/app/src/main/java/ru/anisart/vv/TrackingService.kt L166

Missing Permissions&#10;Call requires permission which may be rejected by user: code should explicitly check to see if permission is available (with &#96;checkPermission&#96;) or explicitly handle a potential &#96;SecurityException&#96;&#10;This check scans through your code and libraries and looks at the APIs being used, and checks this against the set of permissions required to access those APIs. If the code using those APIs is called at runtime, then the program will crash.&amp;#xA;&amp;#xA;Furthermore, for permissions that are revocable (with targetSdkVersion 23), client code must also be prepared to handle the calls throwing an exception if the user rejects the request for permission at runtime.


*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a17914559>*